packages/features/insights/server/routing-events.ts

```diff
@@ -635,6 +645,483 @@ class RoutingEventsInsights {
+        JOIN "App_RoutingForms_Form" f ON r."formId" = f.id
+        WHERE b."userId" IN (SELECT user_id FROM all_users)
+        AND date_trunc(${dayjsPeriod}, b."createdAt") >= (SELECT MIN(period_start) FROM paginated_periods)
+        AND date_trunc(${dayjsPeriod}, b."createdAt") <= (SELECT MAX(period_start) FROM paginated_periods)
+        ${whereClause}
+        GROUP BY 1, 2
+      )
+      -- Join everything together
+      SELECT 
+        c."userId",
+        c.period_start,
+        COALESCE(b.total, 0)::integer as total
+      FROM date_user_combinations c
+      LEFT JOIN booking_counts b ON 
+        b."userId" = c."userId" AND 
+        b.period_start = c.period_start
+      ORDER BY c.period_start ASC, c."userId" ASC
+    `;
+
+    // Get total number of periods to determine if there are more
+    const totalPeriodsQuery = await prisma.$queryRaw<[{ count: number }]>`
+      WITH RECURSIVE date_range AS (
+        SELECT date_trunc(${dayjsPeriod}, ${startDate}::timestamp) as date
+        UNION ALL
+        SELECT date + (CASE 
+          WHEN ${dayjsPeriod} = 'day' THEN interval '1 day'
+          WHEN ${dayjsPeriod} = 'week' THEN interval '1 week'
+          WHEN ${dayjsPeriod} = 'month' THEN interval '1 month'
+        END)
+        FROM date_range
+        WHERE date < date_trunc(${dayjsPeriod}, ${endDate}::timestamp + interval '1 day')
+      )
+      SELECT COUNT(*)::integer as count FROM date_range
+    `;
+
+    // Get statistics for the entire period for comparison
+    const statsQuery = await prisma.$queryRaw<
+      Array<{
+        userId: number;
+        total_bookings: number;
+      }>
+    >`
+      SELECT 
+        b."userId",
+        COUNT(*)::integer as total_bookings
+      FROM "App_RoutingForms_FormResponse" r
+      JOIN "App_RoutingForms_Form" f ON r."formId" = f.id
+      JOIN "Booking" b ON r."routedToBookingUid" = b.uid
+      WHERE r."routedToBookingUid" IS NOT NULL
+      AND r."createdAt" >= ${startDate}
+      AND r."createdAt" <= ${endDate}
+      ${whereClause}
+      GROUP BY b."userId"
+      ORDER BY total_bookings ASC
+    `;
+
+    // Calculate average and median
+    const average =
+      statsQuery.reduce((sum, stat) => sum + Number(stat.total_bookings), 0) / statsQuery.length || 0;
+    const median = statsQuery[Math.floor(statsQuery.length / 2)]?.total_bookings || 0;
+
+    // Create a map of user performance indicators
+    const userPerformance = statsQuery.reduce((acc, stat) => {
+      acc[stat.userId] = {
+        total: stat.total_bookings,
+        performance:
+          stat.total_bookings > average
+            ? "above_average"
+            : stat.total_bookings === median
+            ? "median"
+            : stat.total_bookings < average
+            ? "below_average"
+            : "at_average",
+      };
+      return acc;
+    }, {} as Record<number, { total: number; performance: "above_average" | "at_average" | "below_average" | "median" }>);
```

Comment:
I couldnt do figure out how to do this in SQL so we calculate the avergae for the period per user here and add it in to each row